boards/msp_exp432p401r/src/main.rs

```diff
@@ -409,8 +410,7 @@ pub unsafe fn main() {
             &mut capsules::adc::ADC_BUFFER3
         )
     );
-
-    peripherals.adc.set_client(adc);
+    hil::adc::AdcHighSpeed::set_client(&peripherals.adc, adc);
```

Comment:
if someone were to call `peripherals.adc.set_client(adc)`, does rustc throw a warning that this is ambiguous and could call one of two methods, or does it just pick one based off of what is in scope? I suspect the latter. If so, this still seems to make it easy to set a non-high-speed client on accident, since this is one of very few places in main.rs where you have to fully specify which trait you are using in order to get correct behavior.